apps/prairielearn/src/sprocs/check_assessment_access.sql

```diff
@@ -26,114 +27,165 @@ DECLARE
     active_access_rule_id bigint;
     next_active_start_date TIMESTAMP WITH TIME ZONE;
     next_active_credit integer;
+    credit_from_override integer;
+    assessment_end_date TIMESTAMP WITH TIME ZONE;
+    end_date_from_override TIMESTAMP WITH TIME ZONE;
+    start_date_from_override TIMESTAMP WITH TIME ZONE;
+
 BEGIN
-    -- Choose the access rule which grants access ('authorized' is TRUE), if any, and has the highest 'credit'.
-    SELECT
-        caar.authorized,
-        caar.exam_access_end,
-        aar.credit,
-        CASE
-            WHEN aar.credit > 0 AND aar.active THEN
-                aar.credit::text || '%'
-                || (CASE
-                        WHEN aar.end_date IS NOT NULL
-                        THEN ' until ' || format_date_short(aar.end_date, display_timezone)
-                        ELSE ''
-                    END)
-            ELSE 'None'
-        END AS credit_date_string,
-        -- If timer hits 0:00 at end_date, exam might end after end_date (overdue submission).
-        -- Resolve race condition by subtracting 31 sec from end_date.
-        -- Use 31 instead of 30 to force rounding (time_limit_min is in minutes).
-        CASE WHEN aar.time_limit_min IS NULL THEN NULL
-             WHEN aar.mode = 'Exam' THEN NULL
-             ELSE LEAST(aar.time_limit_min, DATE_PART('epoch', aar.end_date - now() - INTERVAL '31 seconds') / 60)::integer
-        END AS time_limit_min,
-        aar.password,
-        aar.mode,
-        aar.seb_config,
-        aar.show_closed_assessment,
-        aar.show_closed_assessment_score,
-        aar.active,
-        aar.id
-    INTO
-        authorized,
-        exam_access_end,
-        credit,
-        credit_date_string,
-        time_limit_min,
-        password,
-        mode,
-        seb_config,
-        show_closed_assessment,
-        show_closed_assessment_score,
-        active,
-        active_access_rule_id
-    FROM
-        assessment_access_rules AS aar
-        JOIN LATERAL check_assessment_access_rule(aar, check_assessment_access.authz_mode,
-            check_assessment_access.user_id, check_assessment_access.uid, check_assessment_access.date, TRUE) AS caar ON TRUE
-    WHERE
-        aar.assessment_id = check_assessment_access.assessment_id
-        AND caar.authorized
-        AND ((aar.role > 'Student') IS NOT TRUE)
-    ORDER BY
-        aar.credit DESC NULLS LAST,
-        aar.number
+    -- Check if the user has an entry in the assessment_access_policies table for this assessment_id.
+    -- If yes, get the end_date from the assessment_access_policies table, otherwise, use the end_date from the assessment_access_rules table.
```

Comment:
@mwest1066 to make this easier to parse + review, I'm wondering if we should change the approach here and fully handle overrides at the start of the sproc and early-return if there is one.

Should we support things like `show_closed_assessment`, `show_closed_assessment_score`, `time_limit_min`, `password`, etc. here? If not, what default values should we pull for the first three, if any?

Reply:
Added early return in 9b25ab775

Reply:
However, I think that their is additions throughout the file, so if the logic (specifically my declaration of access_rules) I can pull this file from master and add the early return so the diff is clean

Reply:
Ended up pulling from master and fixing - f7fc9b08d

Reply:
Thanks, this is incredibly easier to review and reason about!